src/viem/saga.ts

```diff
@@ -74,53 +82,117 @@ export function* sendPayment({
         comment,
         status: TransactionStatus.Pending,
         value: amount.negated().toString(),
-        tokenAddress,
-        tokenId: getTokenId(networkConfig.defaultNetworkId, tokenAddress),
+        tokenAddress: tokenInfo.address ?? undefined,
+        tokenId,
         timestamp: Math.floor(Date.now() / 1000),
         address: recipientAddress,
       })
     )
+  }
 
-    const receipt = yield* call(sendAndMonitorTransaction, {
-      context,
-      wallet,
-      // Cast for now otherwise TS complains about the type of request
-      // TODO: investigate more and fix
-      request: request as SimulateContractReturnType['request'],
-    })
-    return receipt
+  const tokenAddress = tokenInfo.address
+
+  // For tokens with an address, we simulate calling 'transfer' on the contract,
+  // take the request generated by that simulation, and execute that request
+  //
+  // For tokens with no address, we perform a simple `call` to test the request.
+  try {
+    if (tokenAddress) {
+      if (!feeInfo) {
+        throw new Error('non-native viem sends require feeInfo')
+      }
+      // this returns a method which is then passed to call instead of directly
+      // doing yield* call(publicClient.celo.simulateContract, args) because this
+      // results in a long TS error
+      const simulateContractMethod = yield* call(getTransferSimulateContract, {
+        wallet,
+        tokenId,
+        amount,
+        recipientAddress,
+        comment,
+        feeInfo,
+      })
+
+      const { request } = yield* call(simulateContractMethod)
+      yield* call(unlockAndAddStandby)
+
+      const sendContractTxMethod = function* () {
+        return yield* call(
+          [wallet, 'writeContract'],
+          request as SimulateContractReturnType['request']
+        )
+      }
+
+      const receipt = yield* call(sendAndMonitorTransaction, {
+        context,
+        network,
+        sendTx: sendContractTxMethod,
+      })
+
+      return receipt
+    } else {
+      const convertedAmount = BigInt(
+        yield* call(tokenAmountInSmallestUnit, amount, tokenInfo.tokenId)
+      )
+
+      // The assumption here is that if the call throws, then the actual execution would result
+      // in a revert. I'm not positive that this assumption holds, however.
+      // TODO (ACT-922): Fill in fee estimation for native token transfers (non-Celo)
+      const callMethod = async () =>
+        await publicClient[network].call({
+          account: wallet.account,
+          to: getAddress(recipientAddress),
+          data: '0x00',
+          value: convertedAmount,
+        })
+      Logger.info(TAG, 'Call completed successfully')
+      const callData = yield* call(callMethod)
+      Logger.info(TAG, `calldata: ${JSON.stringify(callData, null, 4)}`)
+      yield* call(unlockAndAddStandby)
+
+      const sendNativeTxMethod = function* () {
+        try {
+          const hash = yield* call([wallet, 'sendTransaction'], {
```

Comment:
So, I've seen some very weird behavior here. On Sepolia, this reliably results in a 503 from the RPC node, with no real indication as to what the actual error is. I've tried using different Sepolia RPC nodes (by using a custom URL in the `transport` field for the Ethereum client in `src/viem/index.ts`) but received similar errors. 

The address that I've been using for testing _also_ has a balance on Goerli. Out of curiosity, I reconfigured the Ethereum client to use the Goerli chain with the default RPC node, and it actually worked, and successfully sent a TX [here](https://goerli.etherscan.io/tx/0xe0fdffc41dc13b6b24f9dd7c19fc9751d961fe40e750920549c854c9be611b99). This is really confusing to me -- I'm not sure what/if there's a meaningful difference between these two testnets.